data/examples/declaration/value/function/operator-chains-out.hs

```diff
@@ -0,0 +1,245 @@
+module A (a) where
```

Comment:
This file is too long. Whenever something fails in it it'll be very hard to locate the problem. Split it into several files, once per case.

Reply:
(Answering to your comment about why ":>" and ":<|>" can't be ordered, but for some reason in the Github interface I can only answer to this (just next) comment)

This is what I get from my hoogle/hackage extraction:
```haskell
      ( ":>",
        -- infixl 5 (7876/24508 = 32.1%) in lens (2096/24508 = 8.6%), containers (1489/24508 = 6.1%), fingertree (790/24508 = 3.2%), diagrams-lib (753/24508 = 3.1%), optics-core (691/24508 = 2.8%), rio (626/24508 = 2.6%), optics-extra (598/24508 = 2.4%), preamble (259/24508 = 1.1%), hgeometry-combinatorial (148/24508 = 0.6%), microformats2-parser (65/24508 = 0.3%), sized (58/24508 = 0.2%), hw-fingertree-strict (46/24508 = 0.2%), hw-fingertree (46/24508 = 0.2%), sequence (45/24508 = 0.2%), sitepipe (35/24508 = 0.1%), knit-haskell (34/24508 = 0.1%), lens-utils (29/24508 = 0.1%), strict-containers (22/24508 = 0.1%), raw-feldspar (19/24508 = 0.1%), planet-mitchell (12/24508 = 0.0%), polysemy-optics (10/24508 = 0.0%), kmonad (5/24508 = 0.0%)
        FixityInfo {fixDir = Just InfixL, fixMinPrec = 5, fixMaxPrec = 5}
        -- infixl 9 (6002/24508 = 24.5%) in futhark (2353/24508 = 9.6%), opaleye (1589/24508 = 6.5%), parsley-core (535/24508 = 2.2%), quickcheck-state-machine (244/24508 = 1.0%), parameterized-utils (231/24508 = 0.9%), jukebox (194/24508 = 0.8%), streams (188/24508 = 0.8%), wumpus-basic (104/24508 = 0.4%), parsley (102/24508 = 0.4%), haskus-utils-data (82/24508 = 0.3%), feldspar-language (69/24508 = 0.3%), presburger (60/24508 = 0.2%), liboleg (57/24508 = 0.2%), type-aligned (52/24508 = 0.2%), hobbits (32/24508 = 0.1%), yices (26/24508 = 0.1%), kansas-lava (22/24508 = 0.1%), linear-maps (15/24508 = 0.1%), serv (14/24508 = 0.1%), data-sketches-core (13/24508 = 0.1%), exposed-containers (7/24508 = 0.0%), spine (5/24508 = 0.0%), VecN (5/24508 = 0.0%), fingertree-tf (3/24508 = 0.0%), language-sh (?/24508 = ?%), joinlist (?/24508 = ?%), darcs-beta (?/24508 = ?%), clean-unions (?/24508 = ?%), ForSyDe (?/24508 = ?%)
          <> FixityInfo {fixDir = Just InfixL, fixMinPrec = 9, fixMaxPrec = 9}
        -- infixr 5 (4971/24508 = 20.3%) in clash-prelude (2290/24508 = 9.3%), massiv (1047/24508 = 4.3%), streaming (680/24508 = 2.8%), generic-lens-core (622/24508 = 2.5%), biohazard (114/24508 = 0.5%), sdp (85/24508 = 0.3%), ribbit (63/24508 = 0.3%), forsyde-shallow (41/24508 = 0.2%), linear-base (29/24508 = 0.1%)
          <> FixityInfo {fixDir = Just InfixR, fixMinPrec = 5, fixMaxPrec = 5}
        -- infixr 4 (2238/24508 = 9.1%) in servant (1330/24508 = 5.4%), prolude (543/24508 = 2.2%), gogol-core (183/24508 = 0.7%), servant-js (147/24508 = 0.6%), ocaml-export (35/24508 = 0.1%)
          <> FixityInfo {fixDir = Just InfixR, fixMinPrec = 4, fixMaxPrec = 4}
        -- infixr 3 (1418/24508 = 5.8%) in HaXml (621/24508 = 2.5%), pandora (599/24508 = 2.4%), PrimitiveArray (161/24508 = 0.7%), joint (37/24508 = 0.2%)
          <> FixityInfo {fixDir = Just InfixR, fixMinPrec = 3, fixMaxPrec = 3}
        -- infix 0 (948/24508 = 3.9%) in extensible (879/24508 = 3.6%), membership (69/24508 = 0.3%)
          <> FixityInfo {fixDir = Just InfixN, fixMinPrec = 0, fixMaxPrec = 0}
        -- infixr 1 (342/24508 = 1.4%) in darcs (342/24508 = 1.4%)
          <> FixityInfo {fixDir = Just InfixR, fixMinPrec = 1, fixMaxPrec = 1}
        -- infixr 0 (163/24508 = 0.7%) in lorentz (80/24508 = 0.3%), llvm-pretty (56/24508 = 0.2%), indigo (27/24508 = 0.1%)
          <> FixityInfo {fixDir = Just InfixR, fixMinPrec = 0, fixMaxPrec = 0}
        -- infixr 6 (161/24508 = 0.7%) in type-of-html (161/24508 = 0.7%)
          <> FixityInfo {fixDir = Just InfixR, fixMinPrec = 6, fixMaxPrec = 6}
        -- infixr 9 (158/24508 = 0.6%) in tfp (68/24508 = 0.3%), sequence (45/24508 = 0.2%), sandwich (45/24508 = 0.2%)
          <> FixityInfo {fixDir = Just InfixR, fixMinPrec = 9, fixMaxPrec = 9}
        -- infix 6 (56/24508 = 0.2%) in type-combinators (56/24508 = 0.2%)
          <> FixityInfo {fixDir = Just InfixN, fixMinPrec = 6, fixMaxPrec = 6}
        -- infix 4 (51/24508 = 0.2%) in ghc-typelits-presburger (51/24508 = 0.2%)
          <> FixityInfo {fixDir = Just InfixN, fixMinPrec = 4, fixMaxPrec = 4}
        -- infixl 8 (49/24508 = 0.2%) in zippers (49/24508 = 0.2%)
          <> FixityInfo {fixDir = Just InfixL, fixMinPrec = 8, fixMaxPrec = 8}
        -- infix 5 (39/24508 = 0.2%) in limp (39/24508 = 0.2%)
          <> FixityInfo {fixDir = Just InfixN, fixMinPrec = 5, fixMaxPrec = 5}
        -- infixr 8 (25/24508 = 0.1%) in graphql-api (25/24508 = 0.1%)
          <> FixityInfo {fixDir = Just InfixR, fixMinPrec = 8, fixMaxPrec = 8}
        -- infixr 7 (11/24508 = 0.0%) in hreq-core (11/24508 = 0.0%)
          <> FixityInfo {fixDir = Just InfixR, fixMinPrec = 7, fixMaxPrec = 7}
      ),
      ( ":<|>",
        -- infixr 3 (2238/2317 = 96.6%) in servant (1330/2317 = 57.4%), prolude (543/2317 = 23.4%), gogol-core (183/2317 = 7.9%), servant-js (147/2317 = 6.3%), ocaml-export (35/2317 = 1.5%)
        FixityInfo {fixDir = Just InfixR, fixMinPrec = 3, fixMaxPrec = 3}
        -- infixl 9 (57/2317 = 2.5%) in magicbane (45/2317 = 1.9%), solga (12/2317 = 0.5%)
        --  (threshold reached) <> FixityInfo {fixDir = Just InfixL, fixMinPrec = 9, fixMaxPrec = 9}
        -- infixl 4 (22/2317 = 0.9%) in eved (22/2317 = 0.9%)
        --  (threshold reached) <> FixityInfo {fixDir = Just InfixL, fixMinPrec = 4, fixMaxPrec = 4}
      ),
```

So basically, for `":>"`, `servant` didn't reach the arbitrary threshold (> 90% of the total "popularity") to be considered *The Only Declaration That Matters ™️*, with its little 5.4% of total popularity. As a result, we get a final `FixityInfo {fixDir = Nothing, fixMinPrec = 0, fixMaxPrec = 9}` for this operator, which in simpler terms means "we don't know anything about this operator".
If we had automatic cabal dependency detection, or manual user whitelisting of some packages (like in https://github.com/tweag/ormolu/pull/832), this issue would become almost nonexistant

Reply:
I'm working on this :)